src/main/utility/byte_queue.rs

```diff
@@ -201,7 +201,22 @@ impl ByteQueue {
                 None => break,
             };
 
-            let copied = dst.write(bytes.as_ref())?;
+            let copied = match dst.write(bytes.as_ref()) {
+                Ok(x) => x,
+                // may have been interrupted due to a signal, so try again
+                Err(e) if e.kind() == ErrorKind::Interrupted => continue,
```

Comment:
I'm not sure this is the right place to swallow interrupts and retry. How exactly to we get an `Interrupted` error here? If this is meant to handle the case of an emulated signal, for a handler that was configured with `SA_RESTART`, I think that will be better done at a higher level; maybe in the SysCallHandler. Or is this some internal Shadow-level interrupt and not part of the simulation?

Reply:
I'm not sure what you mean. This is a reader/writer implementation-specific thing. One example is if you're writing to a file and a write syscall returns EINTR, the `dst.write()` method would return `ErrorKind::Interrupted`. You can't just bubble it up since previous iterations of the loop may have copied bytes out of the byte queue and into `dst`, and if you return an error the caller wouldn't know how many bytes have already been copied so you might lose those bytes.

Reply:
Ah right, I'd forgotten we were dealing with a generic Write object here. Yeah, I agree trying again here makes sense.